packages/gatsby/src/utils/start-server.ts

```diff
@@ -0,0 +1,313 @@
+import chokidar from "chokidar"
+
+import webpackHotMiddleware from "webpack-hot-middleware"
+import webpackDevMiddleware from "webpack-dev-middleware"
+import got from "got"
+import webpack from "webpack"
+import express from "express"
+import graphqlHTTP from "express-graphql"
+import graphqlPlayground from "graphql-playground-middleware-express"
+import graphiqlExplorer from "gatsby-graphiql-explorer"
+import { formatError } from "graphql"
+
+import webpackConfig from "../utils/webpack.config"
+import { store } from "../redux"
+import { buildHTML } from "../commands/build-html"
+import { withBasePath } from "../utils/path"
+import report from "gatsby-cli/lib/reporter"
+import launchEditor from "react-dev-utils/launchEditor"
+import cors from "cors"
+import telemetry from "gatsby-telemetry"
+import * as WorkerPool from "../utils/worker/pool"
+import http from "http"
+import https from "https"
+
+import { developStatic } from "../commands/develop-static"
+import withResolverContext from "../schema/context"
+import { websocketManager, WebsocketManager } from "../utils/websocket-manager"
+import { slash } from "gatsby-core-utils"
+import apiRunnerNode from "../utils/api-runner-node"
+import { Express } from "express"
+
+import { Stage, IProgram } from "../commands/types"
+import JestWorker from "jest-worker"
+
+import {
+  startSchemaHotReloader,
+  stopSchemaHotReloader,
+} from "../bootstrap/schema-hot-reloader"
+
+import sourceNodes from "../utils/source-nodes"
+import { createSchemaCustomization } from "../utils/create-schema-customization"
+import { rebuild as rebuildSchema } from "../schema"
+type ActivityTracker = any // TODO: Replace this with proper type once reporter is typed
+
+interface IServer {
+  compiler: webpack.Compiler
+  listener: http.Server | https.Server
+  webpackActivity: ActivityTracker
+  websocketManager: WebsocketManager
+  workerPool: JestWorker
+}
+
+export async function startServer(
```

Comment:
Here I've changed the signature to accept the app and workerPool from outside, rather than creating them here